Integrations/integration-TruSTAR.yml

```diff
@@ -147,6 +182,172 @@ script:
+        }
+
+    def create_url_ec(indicators, url, threshold):
+        if not indicators:
+            return {
+                'DBotScore': {
+                    'Indicator': url,
+                    'Type': 'url',
+                    'Score': 0,
+                    'Vendor': 'TruSTAR',
+                }
+            }
+        trustar_ec = {}
+        url_ec = {}
+        dbot_ec = {}
+        for indicator in indicators:
+            url_ec.update({
+                'Data': indicator['value'],
+            })
+            trustar_ec.update({
+                'Value': indicator['value'],
+                'Whitelisted': indicator['whitelisted'],
+                'Priority': indicator['priorityLevel']
+            })
+            indicator_score = priority_level_to_score(indicator['priorityLevel'])
+            dbot_ec.update({
+                'Indicator': url,
+                'Type': 'url',
+                'Vendor': 'TruSTAR',
+                'Score': 0 if indicator_score == 0 else (2 if threshold > indicator_score else 3)
+            })
+            if threshold <= indicator_score:
+                url_ec.update({
+                    'Malicious': {
+                        'Vendor': 'TruSTAR',
+                        'Description': indicator['priorityLevel']
+                    }
+                })
+        return {
+            outputPaths['dbotscore']: dbot_ec,
+            outputPaths['url']: url_ec,
+            'TruSTAR.URL': trustar_ec
+        }
+
+    def create_domain_ec(indicators, url, threshold):
+        if not indicators:
+            return {
+                'DBotScore': {
+                    'Indicator': url,
+                    'Type': 'domain',
+                    'Score': 0,
+                    'Vendor': 'TruSTAR',
+                }
+            }
+        trustar_ec = {}
+        domain_ec = {}
+        dbot_ec = {}
+        for indicator in indicators:
+            domain_ec.update({
+                'Name': indicator['value'],
+            })
+            trustar_ec.update({
+                'Value': indicator['value'],
+                'Whitelisted': indicator['whitelisted'],
+                'Priority': indicator['priorityLevel']
+            })
+            indicator_score = priority_level_to_score(indicator['priorityLevel'])
+            dbot_ec.update({
+                'Indicator': url,
+                'Type': 'domain',
+                'Vendor': 'TruSTAR',
+                'Score': 0 if indicator_score == 0 else (2 if threshold > indicator_score else 3)
+            })
+            if threshold <= indicator_score:
+                domain_ec.update({
+                    'Malicious': {
+                        'Vendor': 'TruSTAR',
+                        'Description': indicator['priorityLevel']
+                    }
+                })
+        return {
+            outputPaths['dbotscore']: dbot_ec,
+            outputPaths['domain']: domain_ec,
+            'TruSTAR.Domain': trustar_ec
```

Comment:
Please use unique key

Reply:
Unique key?

Reply:
Oh you meant dt check ( i.e. ('Indicator' === url)).
I'll take care of it.